### PR TITLE
Improve color table widget

### PR DESCRIPTION
**Pull request by tcoulange**

### CODE REVIEW

**Comment by pieper** [comment]
Thanks for the contribution 👍 

Can you provide more specific information about the improvements?  CTK has a tradition of good commit messages so if you look through the commit logs you will get an idea of what I mean.

http://www.commontk.org/index.php/Contributing_to_CTK#Git_Commit_Style

File: Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp [hunk 1]
```diff
@@ -403,44 +425,84 @@ void vtkDiscretizableColorTransferChart::SetCurrentControlPointColor(
...
-}
 
-// ----------------------------------------------------------------------------
-double* vtkDiscretizableColorTransferChart::GetDataRange()
-{
-  return this->DataRange;
+#ifdef DEBUG_RANGE
+  std::cout << "DEBUG_RANGE chart data range = " << this->DataRange[0]
+            << " " << this->DataRange[1] << std::endl;
+#endif
```
Comment (finetjul) [inline Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp]: I'm surprised it does not trigger any refresh/computation

File: Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp [hunk 2]
```diff
@@ -403,44 +425,84 @@ void vtkDiscretizableColorTransferChart::SetCurrentControlPointColor(
...
-  double limitRange[2];
-  limitRange[0] = std::min(this->OriginalRange[0], this->DataRange[0]);
-  limitRange[1] = std::max(this->OriginalRange[1], this->DataRange[1]);
+  if (min == this->GetCurrentRange()[0]
+   && max == this->GetCurrentRange()[1])
+  {
+    return;
+  }
+
+  if (this->OriginalRange[0] < this->OriginalRange[1])
```
Comment (finetjul) [inline Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp]: what if they are equal ?

File: Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp [hunk 3]
```diff
@@ -449,6 +511,32 @@ double* vtkDiscretizableColorTransferChart::GetCurrentRange()
...
+   && max == this->OriginalRange[1])
+  {
+    return;
+  }
+
+  this->OriginalRange[0] = min;
+  this->OriginalRange[1] = max;
+
+#ifdef DEBUG_RANGE
+  std::cout << "DEBUG_RANGE original range = " << this->OriginalRange[0]
```
Comment (finetjul) [inline Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp]: we tend to use qDebug() instead of std::cout

**Comment by finetjul** [comment]
Woops, sorry @pieper , I clicked ```Merge``` before I seeing your message.

**Comment by jcfr** [comment]
@tcoulange Thanks for the patch. When you have a chance, could you remove the dead code, use qDebug(), etc .. as suggested by @finetjul 

As suggested by @pieper , It would also be nice to document in the description of this PR what your change did. 

Merci

**Comment by tcoulange** [comment]
Thank you all for the reviews and advices. I updated the description and will make a new PR to fix this one according to your feedback.

File: Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp (hunk 1 above)
Comment (tcoulange) [reply Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp]: The data range is used to rescale or center the current range, which is triggered by the user. I think we don't want to rescale everytime the data range changes (new input for histogram). Tell me if you don't agree.

File: Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp (hunk 2 above)
Comment (tcoulange) [reply Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp]: good catch, thanks!

File: Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp (hunk 1 above)
Comment (finetjul) [reply Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp]: i guess that works too

**Pull request merged**